nucypher/blockchain/eth/agents.py

```diff
@@ -356,24 +357,20 @@ def sample(self, quantity: int, duration: int, additional_ursulas: float = 1.7,
         """
 
         stakers_population = self.get_staker_population()
-        if quantity > stakers_population:
-            raise self.NotEnoughStakers(f'There are {stakers_population} published stakers, need a total of {quantity}.')
+        n_select = math.ceil(quantity * additional_ursulas)  # Select more Ursulas
+        if n_select > stakers_population:
+            raise self.NotEnoughStakers(f'There are {stakers_population} active stakers, need at least {n_select}.')
```

Comment:
Should we clarify that we need a `sample` that may be larger than the quantity specified. As a user I may be confused that I get a `NotEnoughStakers` exception if I specify `n` = 6 and there are 6 `active stakers` that randomly says some value eg. `need at least 9` (assuming `additional_ursulas` is 1.5)

Perhaps our message could be:
```suggestion
            raise self.NotEnoughStakers(f'There are {stakers_population} active stakers - for {quantity} stakers we need a sample size of at least {n_select}.')
```

Reply:
Yes, I think that's a good suggestion. Thanks @derekpierre !

Reply:
Does `additional_ursulas` still make sense as a static value?  For such small m of n values, a proportional, smaller value might be more usable.

Reply:
@derekpierre sorry I didn't include this in #1280. I'll do that in #1259.
@KPrasch Although in the method signature there's a static default, in reality it's always called explicitly with the `BlockchainPolicy.selection_buffer` value as defined in `BlockchainPolicy.sample_essential()`. We however don't configure `BlockchainPolicy.selection_buffer` yet, so for the moment it always uses the default value of `1.5`. That's why #1061 is still open.